apps/files_sharing/lib/OrphanHelper.php

```diff
@@ -54,11 +54,19 @@ public function fileExists(int $fileId): bool {
 	/**
 	 * @return \Traversable<int, array{id: int, owner: string, fileid: int, target: string}>
 	 */
-	public function getAllShares() {
+	public function getAllShares(?string $owner = null, ?string $with = null) {
 		$query = $this->connection->getQueryBuilder();
 		$query->select('id', 'file_source', 'uid_owner', 'file_target')
 			->from('share')
 			->where($query->expr()->in('item_type', $query->createNamedParameter(['file', 'folder'], IQueryBuilder::PARAM_STR_ARRAY)));
+
+		if ($owner) {
+			$query->andWhere($query->expr()->eq('uid_owner', $query->createNamedParameter($owner)));
+		}
+		if ($with) {
+			$query->andWhere($query->expr()->eq('share_with', $query->createNamedParameter($with)));
+		}
```

Comment:
```suggestion
		if ($owner !== null) {
			$query->andWhere($query->expr()->eq('uid_owner', $query->createNamedParameter($owner)));
		}
		if ($with !== null) {
			$query->andWhere($query->expr()->eq('share_with', $query->createNamedParameter($with)));
		}
```
What if my user is name `'0'` ¿